datafusion/src/scalar.rs

```diff
@@ -43,6 +43,8 @@ pub enum ScalarValue {
     Float32(Option<f32>),
     /// 64bit float
     Float64(Option<f64>),
+    /// 128bit decimal, using the i128 to represent the decimal
+    Decimal128(Option<i128>, Option<usize>, Option<usize>),
```

Comment:
🤔  it seems like we always need the precision and scale (to know what type the Decimal is). 

Thus, perhaps this could be changed to be

```suggestion
    Decimal128(Option<i128>, usize, usize),
```

Reply:
nice point.
I almost forgot this.